### ISSUE TITLE
Problems with repetitive key press chains + temporary enable/disable in action chains

### ISSUE THREAD

**Issue by Riku7** [folivoraAI/BetterTouchTool#2264]
I have hotkeys for iTunes that allow me to very quickly get to change either release year or genre.
The event is a chain as follows:

(first, the key that actually triggers the chain)
emulate cmd+L (selects current song)
➜ emulate cmd+i (displays info window on selected song)
➜ emulate tab (to iterate from data entry field to another)
➜ (and then as many repeats for tab as required in order to get to the correct box for year or genre; on me they're a total of 7 and 8 presses respectively)

Okay, this is simple and it works. But for some reason, very often the amount of tab presses is suddenly wrong, as in the cursor goes to a box too far or too early. I open BTT to correct the amount of tab key presses so the amount is right. Works nicely, I get to do some data fixing in iTunes and call it a day.
Later on some other day when the computer has either been to sleep or restarted, the tab amount is wrong again, and the action chains that worked before are landing the cursor to the wrong box. I fix the amount in BTT again so it works, I do my session, then a new day comes and the problem is back. Why should its functionality change spontaneously between sessions?

If I uncheck "enabled" from a few tab press actions in the chain, they don't get disabled from the chain for some reason, they still happen. So in order to remove excess tab presses, I always have to delete those actions from the chain permanently, and when the action falls short in the future again, to restore tab presses, I have to manually create them to the chain again. So the "enabled" tick doesn't seem to be working in this case, and it would be great if you could copy/duplicate existing actions or chains! I have a lot of BTT settings that are similar but a bit different, so whenever you're creating those, it would be much faster to do if you could duplicate an existing one fully or partially, and then do the small modifications that make it the new one. Like using it as a template.

I strictly have not changed the version of iTunes or my operating system because changing something now would break certain functionality, so this tab problem can't be a result of environmental changes. I just updated BTT to 2.405 and it did not change this issue. The update log said that there was some fix on repeating tasks but it didn't seem to apply here, unfortunately.

Type of Mac: Mac Pro 5,1
macOS version: 10.10.5
BetterTouchTool version: 2.405

**Comment by fifafu** [comment]
it's probably some timing issue, but nothing I will be able to fix in BTT.
You can try to add a "delay next action" after the "cmd+i" to make sure the info window is really opened when the first tab is sent.

In general Apple Script should be used to interact with iTunes. There is a big collection available here: https://dougscripts.com/itunes/index.php

**Comment by Riku7** [comment]
But I don't get it: The amount by which the repeat amount goes off is always 3. 3 tab presses short or 3 tab presses excessive. And it alternates systematically, as in whenever you start using it for the first time, it will be wrong. 

[Edit while writing this] hey I actually figured out why this happens! It's not a timing issue. I have certain AppleScripts that temporarily toggle full keyboard access on for interacting with dialog boxes. Some of my scripts are older and at the time I was only figuring it out, and in some places I had failed to make the accessibility toggle back to its previous state at the correct place. So, the toggling would happen when a certain script would be run, it would work as it should, but it would then leave the toggle state to a position that I don't think of as the norm, and whenever I run a script that either exploits selection of objects by using tab or also relies on doing accessibility toggle at the beginning and end of the script, the state will be wrong. 

If "all controls" is selected, there's an extra amount of boxes to iterate though in the iTunes info window, so you would need three extra tab presses. My default state should be the other option.

![screen shot 2018-03-17 at 18 46 39](https://user-images.githubusercontent.com/6833981/37557865-970a83ea-2a13-11e8-9607-18b1e5c7c0c7.jpg)

So, sorry, that's my bad and I'll go for a hunt of toggle scripts that fail to restore the accessibility state after performing a one-off action.
However, my report on the "enabled" check box as a part of an action chain not working still stays. Disabling non-chain shortcuts by unchecking has been working fine, but as a part of a chain like that, it didn't seem to make any difference.

And yeah, I use Doug's scripts quite a bit but I don't think there's one for quickly altering genre or year on the go. Doing this with BTT also has the perk that you don't get app icons jumping in the dock! Technically I guess I could make an AppleScript that just emulates the exact same chain of key presses that I did using BTT, but I believe that there *could* be a slightly higher chance of missing key presses by using that method, unless you introduce delays...
